.github/workflows/pre_release_version_bump.yml

```diff
@@ -0,0 +1,41 @@
+name: Pre-Release Version Bump
+
+on:
+  workflow_dispatch:
+
+jobs:
+  pre_release_version_bump:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Derive versions
+        id: versions
+        run: |
+          old_version=`grep -oP '(?<=__version__ = ").*(?=")' xir/_version.py`
+          new_version=${old_version%-dev}
```

Comment:
Shouldn't we be removing the `-dev` for the new version here? Maybe I just don't understand this syntax

Reply:
Actually, this _does_ remove the `-dev` suffix. Theo had a similar question here: https://github.com/XanaduAI/xir/pull/12#discussion_r812361081.

From https://tldp.org/LDP/abs/html/string-manipulation.html:

> **${string%substring}**
> Deletes shortest match of _$substring_ from _back_ of _$string_.